### PR TITLE
34 avg goals per game

### LINKED ISSUE #34: Avg Goals per Game

### LINKED ISSUE #35: Highest/Lowest Scoring Home team

### PR DESCRIPTION
**Pull request by Steddy1Love**
Close #34 #35  
Please comment and send questions

### CODE REVIEW

File: spec/game_spec.rb [hunk 1]
```diff
@@ -106,4 +106,8 @@
       expect(Game.average_goals_by_season).to eq (expected)
     end
   end
+
+  it 'averages the goals per game' do
+    expect(@stat_tracker.average_goals_per_game).to eq(4.32)
```
Comment (wallacebilly1) [inline spec/game_spec.rb]: Need to change @stat_tracker to Game and this test will pass.

File: lib/game.rb [hunk 2]
```diff
@@ -97,4 +97,11 @@ def self.average_goals_by_season
     end
     goals_per_season
   end
+
+  def self.average_goals_per_game
```
Comment (wallacebilly1) [inline lib/game.rb:101]: Nice, clean method!

File: lib/stat_tracker.rb [hunk 3]
```diff
@@ -18,6 +18,17 @@ def self.from_csv(locations)
     StatTracker.new(games, teams, game_teams)
   end
 
+  def find_team_name_by_id(team_id)
+    Team.find_team_name_by_id(team_id)
+  end
+
+  def highest_scoring_home_team 
+    find_team_name_by_id(GameTeam.avg_scores_per_team_home.max_by {|team_id, goals| goals}.first)
```
Comment (wallacebilly1) [inline lib/stat_tracker.rb:26]: We may want to refactor these down the road, but nice job getting these figured out! Definitely tough methods.

File: lib/gameteam.rb [hunk 4]
```diff
@@ -35,8 +35,29 @@ def self.create_from_csv(game_teams_path)
...
     end
-    game_teams
+    @@game_teams
+  end
+
+  def self.game_teams
+    @@game_teams
+  end
+
+  def self.avg_scores_per_team_home
```
Comment (wallacebilly1) [inline lib/gameteam.rb:68]: There's not test for this in the gameteam_spec file.  I see that there's a test for the integration test in the stat_tracker, but we'll also need one in the gameteam_spec file as well.

File: spec/stat_tracker_spec.rb [hunk 5]
```diff
@@ -16,44 +16,63 @@
...
     expect(goals_by_season.count).to eq 6
   end
 
+  xit '#average_goals_per_game' do
+    goals_per_game_avg = @stat_tracker.average_goals_per_game
+    expect(goals_per_game_avg).to be_an Float
+    expect(goals_per_game_avg).to eq(4.32)
+  end
+
+  it '#highest_scoring_home_team' do
```
Comment (wallacebilly1) [inline spec/stat_tracker_spec.rb:86]: Nice job with these, they passed the Spec Harness test as well.

**Review by wallacebilly1 — CHANGES_REQUESTED** [review]
Overall, really nice work with the methods you built out.  It looks like you didn't pull down main before this PR (correct me if I'm wrong, but it looked to me like it was missing a few bits from the main branch). Probably easier to address those merge conflicts on your end than doing it when we merge your PR.  

Also, if we merge in my last PR (least_tackles), you can pull down from main at that point and get some other small changes that'll help pass a lot of the tests you x'ed out.

File: lib/gameteam.rb [hunk 6]
```diff
@@ -60,6 +65,23 @@ def self.fewest_tackles_by_season(season_id)
     GameTeam.fewest_tackles(tackles_per_team_hash)
   end
 
+  def self.avg_scores_per_team_home
+    team_scores = Hash.new(0)
```
Comment (grantdavis303) [inline lib/gameteam.rb:69]: I like how you went about this method for the highest and lowest home teams!

File: spec/stat_tracker_spec.rb (hunk 5 above)
Comment (grantdavis303) [reply spec/stat_tracker_spec.rb:86]: Nice clean and concise tests Steddy!

**Pull request merged**